scripts/docker/ci/Dockerfile

```diff
@@ -38,21 +38,41 @@ COPY --from=builder /opt/view /opt/view
 COPY --from=builder /etc/profile.d/celeritas_spack_env.sh /etc/profile.d/celeritas_spack_env.sh
 
 # Add core files
-RUN apt-get -yqq update \
- && apt-get -yqq install --no-install-recommends \
-        build-essential \
-        ca-certificates \
-        libc6-dbg \
-        g++ \
-        gcc \
-        ssh \
-        vim \
- && rm -rf /var/lib/apt/lists/*
+RUN if [ "$DOCKERFILE_DISTRO" == "ubuntu" ] ; then \
+  apt-get -yqq update \
+  && apt-get -yqq install --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    libc6-dbg \
+    g++ \
+    gcc \
+    ssh \
+    vim \
+  && rm -rf /var/lib/apt/lists/* ; \
+elif [ "$DOCKERFILE_DISTRO" == "centos" ] ; then \
+  yum update -y \
+  && yum install -y epel-release \
+  && yum update -y \
+  && yum --enablerepo epel groupinstall -y "Development Tools" \
+  && yum --enablerepo epel install -y \
+       gcc-c++ \
+       gcc \
+       vim \
+  && rm -rf /var/cache/yum \
+  && yum clean all ; \
+fi
```

Comment:
If any of the previous command fails, does the script end or try to proceed with the rest of the script?

Reply:
I think it will fail early as desired:
```console
$ podman build --format docker temp
STEP 1/3: FROM base-jammy-cuda11 AS parent
STEP 2/3: RUN true && false && true
Error: building at STEP "RUN true && false && true": while running runtime: exit status 1
# <edit temp/Dockerfile>
$ podman build --format docker temp
STEP 1/3: FROM base-jammy-cuda11 AS parent
STEP 2/3: RUN if true ; then true && false && true ; fi
Error: building at STEP "RUN if true ; then true && false && true ; fi": while running runtime: exit status 1
```

Reply:
right.  docker look at the result of 'RUN' before moving to the next one.